### PR TITLE
chore: execute staticcheck instead of github action by running golang…

### PR DESCRIPTION
**Pull request by eahydra**
…ci-lint

Signed-off-by: Joseph <joseph.t.lee@outlook.com>

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ linters:
     - misspell
     - vet
     - unconvert
+    - staticcheck
```
Comment (jasonliu747) [inline .golangci.yml:83]: Is it the same as the deleted action?

Comment (eahydra) [reply .golangci.yml:83]: Yes same.

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Pull request merged**